wgpu-types/src/lib.rs

```diff
@@ -30,6 +30,61 @@ pub const COPY_BUFFER_ALIGNMENT: BufferAddress = 4;
 /// Alignment all push constants need
 pub const PUSH_CONSTANT_ALIGNMENT: u32 = 4;
 
+/// Implements conversion into a cow.
+pub trait IntoCow<'c> {
+    type CowType: ?Sized + ToOwned;
```

Comment:
We never have a case where we `CowType` is unknown to us. Therefore, we can turn it into a generic parameter:
```rust
pub trait IntoCow<'c, C> {
  fn into_cow(self) -> Cow<'a, C>;
}
```
The benefit of this would be less typing, i.e. `IntoCow<'a, str>` instead of `IntoCow<'a, CowType = str>`, and I don't think there are downsides for our case.